cloudinit/sources/helpers/vultr.py

```diff
@@ -0,0 +1,254 @@
+# Author: Eric Benner <ebenner@vultr.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import copy
+
+from cloudinit import log as log
+from cloudinit import url_helper
+from cloudinit import dmi
+from cloudinit import util
+from cloudinit import net
+from cloudinit.net.dhcp import EphemeralDHCPv4, NoDHCPLeaseError
+from functools import lru_cache
+
+# Get LOG
+LOG = log.getLogger(__name__)
+
+
+@lru_cache()
+def get_metadata(url, timeout, retries, sec_between):
+    # Bring up interface
+    try:
+        with EphemeralDHCPv4(connectivity_url=url):
+            # Fetch the metadata
+            v1 = read_metadata(url, timeout, retries, sec_between)
+    except (NoDHCPLeaseError) as exc:
+        LOG.error("Bailing, DHCP Exception: %s", exc)
+        raise
+
+    v1_json = json.loads(v1)
+    metadata = v1_json
+
+    # This comes through as a string but is JSON, make a dict
+    metadata['vendor-config'] = json.loads(metadata['vendor-config'])
+
+    return metadata
+
+
+# Read the system information from SMBIOS
+def get_sysinfo():
+    return {
+        'manufacturer': dmi.read_dmi_data("system-manufacturer"),
+        'subid': dmi.read_dmi_data("system-serial-number")
+    }
+
+
+# Assumes is Vultr is already checked
+def is_baremetal():
+    if get_sysinfo()['manufacturer'] != "Vultr":
+        return True
+    return False
+
+
+# Confirm is Vultr
+def is_vultr():
+    # VC2, VDC, and HFC use DMI
+    sysinfo = get_sysinfo()
+
+    if sysinfo['manufacturer'] == "Vultr":
+        return True
+
+    # Baremetal requires a kernel parameter
+    if "vultr" in util.get_cmdline():
```

Comment:
**need to change** (but trivial)

```python
if "vultr" in util.get_cmdline().split()
```

that'll make sure you don't match (/me tries to think of a word with 'vultr' in it)..... "covultronic" :wink:.

A bit manufactured, but might as well be safe.